tests/sentry/api/endpoints/test_organization_details.py

```diff
@@ -783,6 +786,56 @@ def test_org_mapping_already_taken(self):
         OrganizationMapping.objects.create(organization_id=999, slug="taken", region_name="us")
         self.get_error_response(self.organization.slug, slug="taken", status_code=409)
 
+    def test_configure_auth_provider(self):
+        old_config = {"domain": "foo.com"}
+        new_config = {"domain": "bar.com"}
+        provider = "google"
+        self.get_success_response(
+            self.organization.slug,
+            method="put",
+            providerName=provider,
+            providerConfig=old_config,
+        )
+        auth_provider = AuthProvider.objects.get(organization_id=self.organization.id)
+        assert auth_provider.provider == provider
+        assert auth_provider.config == {"domains": ["foo.com"], "version": DATA_VERSION}
+
+        self.get_success_response(
+            self.organization.slug,
+            method="put",
+            providerName=provider,
+            providerConfig=new_config,
+        )
+        auth_provider = AuthProvider.objects.get(organization_id=self.organization.id)
+        assert auth_provider.provider == provider
+        assert auth_provider.config == {"domains": ["bar.com"], "version": DATA_VERSION}
+
+    def test_invalid_auth_provider_configuration(self):
+        self.get_error_response(
+            self.organization.slug, method="put", providerName="google", status_code=400
+        )
+        self.get_error_response(
+            self.organization.slug,
+            method="put",
+            providerConfig={"domain": "foo.com"},
+            status_code=400,
+        )
+        self.get_error_response(
+            self.organization.slug,
+            method="put",
+            providerName="not_valid",
+            providerConfig={"domain": "foo.com"},
+            status_code=400,
+        )
+
+        self.get_error_response(
+            self.organization.slug,
+            method="put",
+            providerName="google",
+            providerConfig={"invalid_domain": "foo.com"},
+            status_code=500,
```

Comment:
@schew2381 This shouldn't be a 500 status code. It should be a 400 status code.

```suggestion
            status_code=400,
```

A status code of 500 means there's an uncaught exception thrown from within some code path on the backend side.

The reason why we want a 400 status code is to communicate to the user that they've provide an invalid set of inputs.

You should try to capture the uncaught exception, and then raise a validation error.